libstorage/MemoryTable2.cpp

```diff
@@ -436,6 +437,73 @@ dev::storage::TableData::Ptr MemoryTable2::dumpWithoutOptimize()
     return m_tableData;
 }
 
+void MemoryTable2::parallelGenData(
+    bytes& _generatedData, std::shared_ptr<std::vector<size_t>> _offsetVec, Entries::Ptr _entries)
+{
+    if (_entries->size() == 0)
+    {
+        return;
+    }
+    tbb::parallel_for(tbb::blocked_range<uint64_t>(0, _offsetVec->size() - 1),
+        [&](const tbb::blocked_range<uint64_t>& range) {
+            for (uint64_t i = range.begin(); i < range.end(); i++)
+            {
+                auto entry = (*_entries)[i];
+                auto startOffSet = (*_offsetVec)[i];
+
+                for (auto& fieldIt : *(entry))
+                {
+                    if (isHashField(fieldIt.first))
+                    {
+                        memcpy(
+                            &_generatedData[startOffSet], &fieldIt.first[0], fieldIt.first.size());
+                        startOffSet += fieldIt.first.size();
+
+                        memcpy(&_generatedData[startOffSet], &fieldIt.second[0],
+                            fieldIt.second.size());
+                        startOffSet += fieldIt.second.size();
+                    }
+                }
+                char status = (char)entry->getStatus();
+                memcpy(&_generatedData[startOffSet], &status, sizeof(status));
+            }
+        });
+}
+
+std::shared_ptr<std::vector<size_t>> MemoryTable2::genDataOffset(
+    Entries::Ptr _entries, size_t _startOffset)
+{
+    std::shared_ptr<std::vector<size_t>> dataOffset = std::make_shared<std::vector<size_t>>();
+    dataOffset->push_back(_startOffset);
+    for (size_t i = 0; i < _entries->size(); i++)
+    {
+        auto entry = (*_entries)[i];
+        auto offset = (*dataOffset)[i] + entry->capacity() + 1;
```

Comment:
add some comments to explain why add 1, like others `capacity() + 1`

Reply:
The comment has been added.